src/doc/en/developer/coding_basics.rst

```diff
@@ -153,17 +153,28 @@ Adding new top-level packages below :mod:`sage` should be done
 sparingly.  It is often better to create subpackages of existing
 packages.
 
-Non-Python Sage source code and supporting files can be included in one
-of the following places:
+Non-Python Sage source code and small supporting files can be
+included in one of the following places:
 
 - In the directory of the Python code that uses that file.  When the
   Sage library is installed, the file will be installed in the same
-  location as the Python code. For example,
-  ``SAGE_ROOT/src/sage/interfaces/maxima.py`` needs to use the file
-  ``SAGE_ROOT/src/sage/interfaces/maxima.lisp`` at runtime, so it refers
-  to it as ::
+  location as the Python code. This is referred to as "package data".
 
-    os.path.join(os.path.dirname(__file__), 'sage-maxima.lisp')
+  The preferred way to access the data from Python is using the
+  function :func:`importlib.resources.files`.
+
+  .. NOTE::
+
+     You may notice that some older code in the Sage library accesses
+     the package data in more direct ways. For example,
+     ``SAGE_ROOT/src/sage/interfaces/maxima.py`` uses the file
+     ``SAGE_ROOT/src/sage/interfaces/maxima.lisp`` at runtime, so it
+     refers to it as::
+
+       os.path.join(os.path.dirname(__file__), 'sage-maxima.lisp')
+
+     This is no longer recommended, and PRs that update such uses
+     are welcome.
```

Comment:
We should not un-recommend this until `sage.interfaces` is a package or we remove support for python 3.9. The alternative is _ugly_, and is adding code that will have to be replaced again soon.

Reply:
> adding code that will have to be replaced again soon

why is this?

Reply:
importlib_resources is only needed for namespace packages for python 3.9 so your `try` hack is unecessary with python 3.10.

Reply:
Please. It's not a "hack". These are the best practices.

And by documenting this method to import it, we avoid the "too much information" problem where we tell developers every little detail of what is supported on what version etc.

Reply:
> Please. It's not a "hack". These are the best practices.

According to whom?

Reply:
OK, maybe second-best practices

Reply:
*Now* (with abf3ae6), at best practices.

Reply:
Please give an external reference (i.e. not yourself)

Reply:
Hard to tell what you are looking for. https://importlib-resources.readthedocs.io/en/latest/using.html# leaves no doubt how to import from it. We have decided to conditionalize the dependency on the backport package in #36776 @orlitzky, hence here we conditionalize the import.

In case you are looking for this: You can see how it is used in projects in the wild in this code search:
https://github.com/search?q=%22from+importlib_resources+import%22&type=code&p=1

Reply:
> We should not un-recommend this until ...

@tornaria I've changed it in 4d38922eb81, please take a look.